### ISSUE TITLE
[BUG] Z steppers get disabled when probe Z-offset babystepping during a print

### ISSUE THREAD

**Issue by onfoot** [MarlinFirmware/Marlin#19277]
### Bug Description

Sometimes when modifying the Probe Z offset during printing the first layer, most likely when a retraction hop is happening simultaneously, the steppers in my independent dual-z setup get disengaged and there is no way of enabling them in any other way than by a reset.

### My Configurations

[stepper-bug-configs.zip](https://github.com/MarlinFirmware/Marlin/files/5179566/stepper-bug-configs.zip)

### Steps to Reproduce

1. Start an SD print
2. Watch in panic as nozzle is scraping the build plate or flying slightly above it.
3. Go into probe z-offset screen and start rotating the knob.
4. Hear the thump  of Z motors being disabled, emit a sigh of disappointment.
5. Stop the print.
6. Go into Motion/Move Axis/Z and try to move it around.
7. Notice no movement at all.
8. Be able to rotate the z-screws easily with fingers.

**Expected behavior:** The Z steppers are still powered and act on babystepping, perform retraction hops, and lift the nozzle (or drop the bed) to advance to subsequent layers.

**Actual behavior:** Z steppers are disabled and the z-screws can be moved freely, thus firmware has no control over them any longer.

#### Additional Information

I tried that on latest bugfix-2.0.x (latest commit: d9ad8caf0281398dfe960b4babb4e46512e18b18)

I modified the MKS SGEN L v2 pin definition to enable the second Z stepper (I prefer to be able to autoalign the bed, but will eventually convert the printer to single Z stepper driving both screws synchronized with a belt).
[pins_MKS_SGEN_L_V2.h.zip](https://github.com/MarlinFirmware/Marlin/files/5180014/pins_MKS_SGEN_L_V2.h.zip)

Since I discovered the `BABYSTEP_ZPROBE_OFFSET` I enabled it on my other printer and while doing the adjustments I think that one also disabled the Z stepper at one point, but I'm not sure whether it was also permanent or only while babystepping.

**Comment by sjasonsmith** [comment]
You might have more success with your 2209 drivers on the Z axis. They are less susceptible to this than 2208s.

You may also try enabling the experimental INTEGRATED_BABYSTEPPING.

**Comment by onfoot** [comment]
Oh, so this is some kind of a driver issue?

I'll try the integrated babystepping, thanks for the pointer.

**Comment by sjasonsmith** [comment]
Babystepping adds some unsynchronized stepper pulses into your normal sequence of steps. If your printer is already moving Z this can lead to an overload condition (or something) in TMC drivers, and they disable themselves to save themselves.

This came up before, but I was super busy at the time and didn't see exactly what the resolution was. You can probably search closed issues and find it, probably by searching for INTEGRATED_BABYSTEPPING.

Basically, some stuff was done to try to aid the situation, but I'm not sure how perfect the solution is. I haven't heard anyone mention it for a while.

**Comment by onfoot** [comment]
Yes, found at least two, [this one](https://github.com/MarlinFirmware/Marlin/issues/16617) being the most interesting.

And probably this was done to alleviate the issue to some extent:

```
#if ENABLED(INTEGRATED_BABYSTEPPING)
      if (is_babystep)                                  // Avoid ANY stepping too soon after baby-stepping
        NOLESS(nextMainISR, (BABYSTEP_TICKS) / 8);      // FULL STOP for 125µs after a baby-step

      if (nextBabystepISR != BABYSTEP_NEVER)            // Avoid baby-stepping too close to axis Stepping
        NOLESS(nextBabystepISR, nextMainISR / 2);       // TODO: Only look at axes enabled for baby-stepping
    #endif
```

**Comment by sjasonsmith** [comment]
OK, please try that out and report back how it works. Please don't immediately close this issue if it fixes things, because we may need to discuss how to make this more discoverable for people who need it.

**Comment by onfoot** [comment]
Sure thing. I'll check it out as soon as I can.

**Comment by onfoot** [comment]
After enabling integrated babystepping the issue is still there, I'm afraid. Most of the time when that happens both steppers get disabled. Sometimes, though, only one, which is always fun (and one more argument for moving to a single stepper setup). ;)

After I'm done printing a golden Mimikyu for my kid (yeah, right) I'll try babystepping with StealthChop disabled, as the other issue I linked to suggested.
